File: 08-download-a-file-revisited/csharp/FileDownloadRevisited.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using System.Net;

public class FileDownloadRevisited
{

    IWebDriver Driver;

    [SetUp]
    public void SetUp()
    {
        Driver = new FirefoxDriver();
    }

    [TearDown]
    public void TearDown()
    {
        Driver.Quit();
    }

    [Test]
    public void CheckFileDownloadWithoutBrowser()
    {
        Driver.Navigate().GoToUrl("http://the-internet.herokuapp.com/download");
        string FileURL = Driver.FindElement(By.CssSelector(".example a")).GetAttribute("href");
        var Request = (HttpWebRequest)WebRequest.Create(FileURL);
        Request.Method = "HEAD";
        WebResponse Response = Request.GetResponse();
        Assert.That(Response.ContentType == "application/octet-stream");
        Assert.That(Response.ContentLength > 0);
    }
}

File: 08-download-a-file-revisited/python/file_download_revisited.py
# -*- coding: utf-8 -*-
"""
Implementation of http://elementalselenium.com/tips/8-download-a-file-revisited
"""

import unittest
from selenium import webdriver
import http.client # Use httplib if using Python 2.x.x


class FileDownloadRevisited(unittest.TestCase):

    def setUp(self):
        self.driver = webdriver.Firefox()

    def tearDown(self):
        self.driver.quit()

    def test_example_1(self):
        driver = self.driver
        driver.get('http://the-internet.herokuapp.com/download')
        download_link = driver.find_element_by_css_selector('.example a').get_attribute('href')

        connection = http.client.HTTPConnection('the-internet.herokuapp.com')
        connection.request('HEAD', download_link)
        response = connection.getresponse()
        connection.close()
        content_type = response.getheader('Content-type')
        content_length = int(response.getheader('Content-length'))

        assert content_type == 'application/octet-stream'
        assert content_length > 0

if __name__ == "__main__":
    unittest.main()
